#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Define _exit with attribute_hidden.
# DP: Related bugs: 
# DP: Author: Roland McGrath
# DP: Upstream status: In CVS
# DP: Status Details: 
# DP: Date: 2003-12-31

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

2003-12-21  Roland McGrath  <roland@frob.com>

      * sysdeps/mach/hurd/dl-sysdep.c (_exit): Define with
      attribute_hidden.

# append the patch here and adjust the -p? flag in the patch calls.
Index: sysdeps/mach/hurd/dl-sysdep.c
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/mach/hurd/dl-sysdep.c,v
retrieving revision 1.74
retrieving revision 1.75
diff -u -p -r1.74 -r1.75
--- sysdeps/mach/hurd/dl-sysdep.c	28 Sep 2002 19:45:28 -0000	1.74
+++ sysdeps/mach/hurd/dl-sysdep.c	22 Dec 2003 06:58:09 -0000	1.75
@@ -1,5 +1,6 @@
 /* Operating system support for run-time dynamic linker.  Hurd version.
-   Copyright (C) 1995,96,97,98,99,2000,01,02 Free Software Foundation, Inc.
+   Copyright (C) 1995,1996,1997,1998,1999,2000,2001,2002,2003
+	Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -571,7 +572,7 @@ __getcwd (char *buf, size_t size)
   return NULL;
 }
 
-void weak_function
+void weak_function attribute_hidden
 _exit (int status)
 {
   __proc_mark_exit (_dl_hurd_data->portarray[INIT_PORT_PROC],
